pkg/ddl/create_table.go

```diff
@@ -330,6 +330,11 @@ func onCreateView(jobCtx *jobContext, job *model.Job) (ver int64, _ error) {
 			return ver, errors.Trace(err)
 		}
 	}
+	if oldTableID > 0 && !orReplace {
```

Comment:
the branch at L325 is the handling of this case, but set at the wrong location https://github.com/pingcap/tidb/pull/10170/files#diff-9dd5a0976146b94bb3074ca04e8131dffc03096505755440e64760b42bbde0fdR106-R110

Reply:
err is nil, the code does not go to the branch.  @D3Hunter

Reply:
yes, i mean you can set the err to TableAlreadyExist if tableID>0, then that branch will work. currently that branch is a dead one

Reply:
You mean this?

```
if oldTableID > 0 && err == nil {
   err = infoschema.ErrTableExists
}
```

What's the benefit over current code? @D3Hunter

Reply:
L325 is a dead branch now, I hope in your PR you can fix it, or make use of it